tools/nanos_gdb.py

```diff
@@ -0,0 +1,69 @@
+import gdb
+
+# these definitions and tagof must be in sync with C source
+class Tag:
+    unknown = 0
+    string = 1
+    symbol = 2
+    table_tuple = 3
+    function_tuple = 4
+
+def tagof(p):
+    return ((p.cast(typ('u64')) >> 38) & 7)
```

Comment:
VA_TAG_OFFSET in src/x86_64/machine.h is defined as 39, shouldn't there be 39 here as well?

Reply:
VA_TAG_OFFSET changed from 39 to 38 with the merging of the linear_backed_heap code. This is due to bit 46 being used to indicate a linear-backed mapping.